reactor-netty-http/src/main/java/reactor/netty/http/server/HttpTrafficHandler.java

```diff
@@ -507,10 +509,19 @@ protected void channelIdle(ChannelHandlerContext ctx, IdleStateEvent evt) {
 		static void addIdleTimeoutHandler(ChannelPipeline pipeline, @Nullable Duration idleTimeout) {
 			if (idleTimeout != null) {
 				String baseName = pipeline.get(NettyPipeline.HttpCodec) != null
-						? NettyPipeline.HttpCodec : NettyPipeline.H2CUpgradeHandler;
+						? NettyPipeline.HttpCodec : pipeline.get(NettyPipeline.H2CUpgradeHandler) != null
```

Comment:
Wdyt if we use if/else so that it is more obvious what's happening?

Reply:
Sure!

Reply:
Done!